cidrgrep.go

```diff
@@ -0,0 +1,31 @@
+package cidrgrep
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"regexp"
+)
+
+// Filter filters input for lines containing an IP within cidr
```

Comment:
Here's a good opportunity for us to really document like a pro. A good doc comment for a function says what its inputs are, what its outputs are, and what happens in between. So here, for example, we might say something like:

```go
// Filter reads from its input reader, looking for lines containing IP addresses
// which match the supplied pattern, which is a CIDR network address expression
// as accepted by net.ParseCIDR. Any matching lines are written to the output
// writer. All other input is ignored. The input reader is read to completion,
// or the first read error.
```

Reply:
Done. I added a new parameter for prefixing output like grep does when searching multiple files so expanded for that too